--- base.py.125	2012-10-10 12:57:52.000000000 +0200
+++ base.py	2012-10-10 12:58:39.000000000 +0200
@@ -234,7 +234,7 @@
         line = soutint.readline().strip()
         if line:
             print line
-            if line.find('Go to ') > -1: # This line is written out by the test runner to system.out and is not log file dependent
+            if line.find('Listening for HTTP') > -1:
                 soutint.close()
                 break
 
